lib/inc/drogon/Cookie.h

```diff
@@ -307,20 +307,19 @@ class DROGON_EXPORT Cookie
      * str2. so the function doesn't apply tolower to the second argument
      * str2 as it's always in lower case.
      *
-     * @return 0 if both strings are equall ignoring case or the difference
-     * between their mismatching characters in lower case.
+     * @return 0 if both strings are equall ignoring case, negative value if lhs
+     * is smaller than rhs and vice versa
      */
     static int stricmp(const string_view &str1, const string_view &str2)
```

Comment:
We shouldn't take a reference of `string_view` since copy for `string_view` is sufficiently optimized and reference can prevent compiler optimization.

```cpp
static int stricmp(string_view str1, string_view str2)
```

We can do the same much more simply: (make sure to include corresponding headers, e.g., `algorithm`)

```cpp
return std::equal(
    str1.cbegin(), str1.cend(), str2.cbegin(), str.cend(),
    [](char c1, char c2) {
        return std::tolower(c1) == c2;
    });
```

Reply:
Also, I would place this function in `lib/inc/drogon/utils/Utilities.h` since it does not need to be specialized for this in terms of asymptotic complexity. Just similar to LLVM implementation (https://llvm.org/doxygen/classllvm_1_1StringRef.html#ae46058c90a3c703357331a6501b32f1c), we get:

```cpp
inline bool equals_insensitive(string_view lhs, string_view rhs)
{
    return std::equal(
        lhs.cbegin(), lhs.cend(), rhs.cbegin(), rhs.cend(),
        [](char lhc, char rhc) {
            return std::tolower(lhc) == std::tolower(rhc);
        });
}
```

Reply:
Not specializing it leads to calling `tolower` on every char of the second argument which can be written in small case beforehand.
I know that's not much but that what @an-tao recommended, so what should I do?

Reply:
I know now that returning an int that can be negative is used for ordering purposes which doesn't seem to important in this case, but other than looking simpler does it really matter?